elastic/kibana
# [Response Ops] Calcuating `schedule_delay` using task `retryAt` when appropriate.

Resolves https://github.com/elastic/kibana/issues/119643

## Summary

* Updated the task manager `updateByQuery` script to set the task `scheduledAt` field to either `runAt` or `retryAt` depending which time has passed. This represents the time at which the task was considered claimable.
  * The `scheduledAt` field was originally set only when the task document is created (so when a rule was created or updated) and it is never updated after that. I didn't see it being used anywhere in the code so it seems like a good candidate for storing this information
* Updated the `scheduleDelay` calculation to use `scheduledAt` instead of `runAt` for comparison.

## To verify

* create a rule that always times out
* verify that the `schedule_delay` inside the event log `execute` action does not keep increasing with every execution. Ideally in a healthy system this delay is around the task manager polling interval of `3s`

### Checklist
